### PR TITLE
add bonus episode: publish docker image github package

### LINKED ISSUE #53: Bonus Episode: Publishing docker images with github packages
This would be similar to the https://github.com/hsf-training/hsf-training-singularity-webpage/issues/13

### PR DESCRIPTION
**Pull request by amanmdesai**
Closes #53

### CODE REVIEW

File: _episodes/12-bonus.md [hunk 1]
```diff
@@ -0,0 +1,114 @@
...
+
+Python packages can be installed using a Docker image. The following example illustrates how to write a Dockerfile for building an image containing python packages.
+
+```text
+FROM ubuntu:20.04
+
+ENV DEBIAN_FRONTEND=noninteractive
+
+RUN apt-get update \
+ && apt-get install -yq \
```
Comment (klieret) [inline _episodes/12-bonus.md]: I don't understand this line: Isn't it missing the package to-be-installed?

File: _episodes/12-bonus.md [hunk 2]
```diff
@@ -0,0 +1,114 @@
...
+ && apt-get update -y \
```
Comment (klieret) [inline _episodes/12-bonus.md]: There was already an `update` in the first line

File: _episodes/12-bonus.md [hunk 3]
```diff
@@ -0,0 +1,114 @@
...
+
+## Publish Docker images with GitHub Packages and share them!
+
+It is possible to publish Docker images with [GitHub packages](https://github.com/features/packages).
+To do so, one needs to use GitHub CI/CD. A step-by-step guide is presented here.
+
+* **Step 1**: Create a GitHub repository and clone it locally.
+* **Step 2**: In the empty repository, make a folder called `.github/workflows`. In this folder we will store the file containing the YAML script for a GitHub workflow, named `Docker-build-deploy.yml` (the name doesn't really matter).
+* **Step 3**: In the top directory of your GitHub repository, create a file named `Dockerfile`.
+* **Step 4**: Copy-paste the content above and add to the Dockerfile. (In principle it is possible to build this image locally, but we will not do that here, as we wish to build it with GitHub CI/CD).
```
Comment (klieret) [inline _episodes/12-bonus.md]: "add it"

File: _episodes/12-bonus.md [hunk 4]
```diff
@@ -0,0 +1,114 @@
+---
+title: "Bonus Episode"
+teaching: 40
+exercises: 0
+questions:
+- How to build Docker container for python packages?
```
Comment (klieret) [inline _episodes/12-bonus.md]: "a Docker container"?

**Review by klieret — APPROVED** [review]
This is a great addition! I only have some minor comments

File: _episodes/12-bonus.md (hunk 4 above)
Comment (amanmdesai) [reply _episodes/12-bonus.md]: done

File: _episodes/12-bonus.md (hunk 1 above)
Comment (amanmdesai) [reply _episodes/12-bonus.md]: that was an unnecessary repetition. I have removed it. Thanks.

File: _episodes/12-bonus.md (hunk 2 above)
Comment (amanmdesai) [reply _episodes/12-bonus.md]: removed

File: _episodes/12-bonus.md (hunk 3 above)
Comment (amanmdesai) [reply _episodes/12-bonus.md]: done

**Comment by amanmdesai** [comment]
Hi, thanks for your comments. I have implemented your comments.

**Review by klieret — APPROVED** [review]
Thanks again! ❤️

**Comment by klieret** [comment]
@all-contributors please add @amanmdesai for content

**Pull request merged**